src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,118 @@
 package core.basesyntax;
 
+import java.util.Arrays;
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+
+    private static final int DEFAULT_SIZE = 10;
+    private Object[] arrayList = new Object[DEFAULT_SIZE];
+    private int size = 0;
+
+    private void grow() {
+        arrayList = Arrays.copyOf(arrayList, (int)(arrayList.length * 1.5));
+    }
+
     @Override
     public void add(T value) {
-
+        if (arrayList.length == size()) {
+            grow();
+        }
+        arrayList[size()] = value;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        if (index < 0 || index > size()) {
+            throw new ArrayListIndexOutOfBoundsException("Index " + index + " out of bounds");
+        }
+        if (size() == arrayList.length) {
+            grow();
+        }
+        Object[] newList = new Object[arrayList.length];
 
+        System.arraycopy(arrayList, 0, newList, 0, index);
+        newList[index] = value;
+        for (int i = index; i < size(); i++) {
+            newList[i + 1] = arrayList[i];
+        }
+        arrayList = newList;
+        size++;
     }
 
     @Override
     public void addAll(List<T> list) {
 
+        if (arrayList.length < size() + list.size()) {
+            grow();
+        }
```

Comment:
The `addAll` method does not correctly handle the case where the combined size of the current list and the list to be added exceeds the current capacity. Consider using a loop to grow the array until it can accommodate all elements, or calculate the required size and grow the array accordingly.